### PR TITLE
#670　Adding　Molecular Printing

### PR DESCRIPTION
**Pull request by morisanhf**
I checked it in local,so I think it works.
please review this PR.
(Acutually,this is my first commit to OSS, so i may make a mistake in how to commit.

### CODE REVIEW

File: src/Dealer.ts [hunk 1]
```diff
@@ -365,6 +365,7 @@ import { IceMoonColony } from "./cards/colonies/IceMoonColony";
 import { InterplanetaryColonyShip } from "./cards/colonies/InterplanetaryColonyShip";
 import { MiningColony } from "./cards/colonies/MiningColony";
 import { MinorityRefuge } from "./cards/colonies/MinorityRefuge";
+import { MolecularPrinting } from "./cards/colonies/MolecularPrinting";
```
Comment (vincentneko) [inline src/Dealer.ts:368]: You need to add this card to ALL_COLONIES_PROJECTS_CARDS like other colonies project cards.
You have the error message "'MolecularPrinting' is declared but its value is never read." above.

File: src/cards/colonies/MolecularPrinting.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    public name: CardName = CardName.MOLECULAR_PRINTING;
+    public cardType: CardType = CardType.AUTOMATED;
+
+    public play(player: Player, game: Game) {
+        let coloniesCount: number = 0;
+        game.colonies.forEach(colony => { 
+          coloniesCount += colony.colonies.length;
+        }); 
+       
+            player.megaCredits += game.getCitiesInPlay()
```
Comment (vincentneko) [inline src/cards/colonies/MolecularPrinting.ts:21]: You can make a single call here:

`player.megaCredits += game.getCitiesInPlay() + coloniesCount;`

Please also mind the indentations,

**Comment by bafolts** [comment]
This looks close. If you need help finishing this up @morisanhf . Otherwise we will wait for you to make requested changes. Looks great!

**Comment by morisanhf** [comment]
I found I failed in npm run cover test. Where should i have fixed?
Anything wrong with MolecularPrinting.spec.ts?

**Comment by vincentneko** [comment]
> I found I failed in npm run cover test. Where should i have fixed?
> Anything wrong with MolecularPrinting.spec.ts?

I think the MolecularPrinting.spec.ts is fine. The error is related with Turmoil it seems. Is your repo up to date with this one ?

**Comment by morisanhf** [comment]
I’m sorry my repo was not up-to-date.
I Synced my repo with your repo.

**Comment by vincentneko** [comment]
This looks good to me.
@bafolts can you please double check ?

File: src/Dealer.ts [hunk 3]
```diff
@@ -590,6 +591,7 @@ export const ALL_COLONIES_PROJECTS_CARDS: Array<ICardFactory<IProjectCard>> = [
     { cardName: CardName.MARKET_MANIPULATION, factory: MarketManipulation },
     { cardName: CardName.MINING_COLONY, factory: MiningColony },
     { cardName: CardName.MINORITY_REFUGE, factory: MinorityRefuge },
+    { cardName: CardName.MOLECULAR_PRINTING,factory:MolecularPrinting},
```
Comment (pierrehilbert) [inline src/Dealer.ts:594]: Please add a spaces between comma and "factory" and between "factory:" and "MolecularPrinting".
Won't change anything but we be more readable.

File: package.json [hunk 4]
```diff
@@ -6,6 +6,7 @@
   "dependencies": {
     "@istanbuljs/nyc-config-typescript": "^0.1.3",
     "@types/chai": "^4.2.11",
+    "@types/jest": "^25.2.1",
```
Comment (bafolts) [inline package.json:9]: We don't use jest. The modifications to `package.json` and `package-lock.json` in this pull request are not necessary.

**Pull request merged**